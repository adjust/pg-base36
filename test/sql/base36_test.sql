-- base36 conversion should encode integer;
SELECT 120::base36;
-- base36 conversion should decode text;
SELECT '3c'::base36::int;
-- base36 conversion should error to big values;
SELECT '3caaaaaaaaaaaaa'::base36::bigint;
-- base36 conversion should cast big base36 values;
SELECT 'zik0zj'::base36::int;
-- base36 conversion should cast big int values;
SELECT 2147483647::base36;
-- base36 conversion should error to big values;
SELECT 'zik0zk'::base36::int;
-- base36 conversion should error invalid values;
SELECT '3&'::base36::bigint;
-- base36 comparison should compare using >;
SELECT 'a'::base36 > 'b'::base36;
-- base36 comparison should compare using <;
SELECT 'a'::base36 < 'b'::base36;
-- base36 negative values should convert from base36;
SELECT '-zik0zj'::base36::int;
-- base36 negative values should convert from ints;
SELECT '-2147483647'::integer::base36;
