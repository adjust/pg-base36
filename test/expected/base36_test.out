BEGIN;
-- base36 conversion should encode integer;
-- ./spec/base36_spec.rb:9;
CREATE EXTENSION base36;
SELECT 120::base36;
 base36 
--------
 3c
(1 row)

ROLLBACK;
BEGIN;
-- base36 conversion should decode text;
-- ./spec/base36_spec.rb:13;
CREATE EXTENSION base36;
SELECT '3c'::base36::int;
 int4 
------
  120
(1 row)

ROLLBACK;
BEGIN;
-- base36 conversion should error to big values;
-- ./spec/base36_spec.rb:17;
CREATE EXTENSION base36;
SELECT '3caaaaaaaaaaaaa'::base36::bigint;
ERROR:  value "3caaaaaaaaaaaaa" is out of range for type base36
LINE 1: SELECT '3caaaaaaaaaaaaa'::base36::bigint;
               ^
ROLLBACK;
BEGIN;
-- base36 conversion should cast big base36 values;
-- ./spec/base36_spec.rb:21;
CREATE EXTENSION base36;
SELECT 'zik0zj'::base36::int;
    int4    
------------
 2147483647
(1 row)

ROLLBACK;
BEGIN;
-- base36 conversion should cast big int values;
-- ./spec/base36_spec.rb:25;
CREATE EXTENSION base36;
SELECT 2147483647::base36;
 base36 
--------
 zik0zj
(1 row)

ROLLBACK;
BEGIN;
-- base36 conversion should error to big values;
-- ./spec/base36_spec.rb:29;
CREATE EXTENSION base36;
SELECT 'zik0zk'::base36::int;
ERROR:  value "zik0zk" is out of range for type base36
LINE 1: SELECT 'zik0zk'::base36::int;
               ^
ROLLBACK;
BEGIN;
-- base36 conversion should error invalid values;
-- ./spec/base36_spec.rb:33;
CREATE EXTENSION base36;
SELECT '3&'::base36::bigint;
ERROR:  value "&" is not a valid digit for type base36
LINE 1: SELECT '3&'::base36::bigint;
               ^
ROLLBACK;
BEGIN;
-- base36 comparison should compare using >;
-- ./spec/base36_spec.rb:39;
CREATE EXTENSION base36;
SELECT 'a'::base36 > 'b'::base36;
 ?column? 
----------
 f
(1 row)

ROLLBACK;
BEGIN;
-- base36 comparison should compare using <;
-- ./spec/base36_spec.rb:43;
CREATE EXTENSION base36;
SELECT 'a'::base36 < 'b'::base36;
 ?column? 
----------
 t
(1 row)

ROLLBACK;
BEGIN;
-- base36 negative values should convert from base36;
-- ./spec/base36_spec.rb:49;
CREATE EXTENSION base36;
SELECT '-zik0zj'::base36::int;
    int4     
-------------
 -2147483647
(1 row)

ROLLBACK;
BEGIN;
-- base36 negative values should convert from ints;
-- ./spec/base36_spec.rb:53;
CREATE EXTENSION base36;
SELECT '-2147483647'::integer::base36;
 base36  
---------
 -zik0zj
(1 row)

ROLLBACK;
